php/Blockonomics.php

```diff
@@ -238,4 +292,267 @@ public function testSetup()
+        return get_option('blockonomics_nojs', false);
+    }
+
+    // Check if a lite mode setting is activated
+    public function is_lite_mode_active(){
+        return get_option('blockonomics_lite', false);
+    }
+
+    // Adds the header to the blockonomics page
+    public function load_blockonomics_header($template_name){
+        add_action('wp_enqueue_scripts', 'bnomics_enqueue_stylesheets' );
+        // Apply nojs header changes
+        if (!$this->is_nojs_template($template_name)) {
+            add_action('wp_enqueue_scripts', 'bnomics_enqueue_scripts' );
+        }
+        // Apply lite-mode header
+        if($this->is_lite_mode_active()){
+        ?>
+          <link rel="stylesheet" type="text/css" href="<?php echo plugins_url('css/order.css', dirname(__FILE__));?>">
+        <?php
+        }else{
+          get_header();
+        }
+    }
+
+    // Adds the footer to the blockonomics page
+    public function load_blockonomics_footer($template_name){
+        // Apply lite-mode footer changes
+        if($this->is_lite_mode_active()){
+            // Apply nojs footer changes
+            if (!$this->is_nojs_template($template_name)) {
+                ?>
+                  <script>var ajax_object = {ajax_url:"<?php echo admin_url( 'admin-ajax.php' ); ?>", wc_url:"<?php echo WC()->api_request_url('WC_Gateway_Blockonomics'); ?>"};
+                  </script>
+                  <script src="<?php echo plugins_url('js/angular.min.js', dirname(__FILE__));?>"></script>
+                  <script src="<?php echo plugins_url('js/angular-resource.min.js', dirname(__FILE__));?>"></script>
+                  <script src="<?php echo plugins_url('js/app.js', dirname(__FILE__));?>"></script>
+                  <script src="<?php echo plugins_url('js/angular-qrcode.js', dirname(__FILE__));?>"></script>
+                  <script src="<?php echo plugins_url('js/vendors.min.js', dirname(__FILE__));?>"></script>
+                  <script src="<?php echo plugins_url('js/reconnecting-websocket.min.js', dirname(__FILE__));?>"></script>
+                <?php
+            }
+        }else{
+          get_footer();
+        }
+    }
+
+    // Adds the selected template to the blockonomics page
+    public function load_blockonomics_template($template_name){
+        $this->load_blockonomics_header($template_name);
+
+        // Load the selected template
+        // Check if child theme or parent theme have overridden the template
+        $template = 'blockonomics_'.$template_name.'.php';
+        if ( $overridden_template = locate_template( $template ) ) {
+            load_template( $overridden_template );
+        } else {
+            load_template( plugin_dir_path(__FILE__)."../templates/" .$template );
+        }
+
+        $this->load_blockonomics_footer($template_name);
+
+        exit();
+    }
+
+    // Load the the payment confirmed template in the page
+    public function load_payment_confirmation_template(){
```

Comment:
Rename function name that indicates it is for noJS flow

Reply:
Renamed in 9261a5c